tools/run_tests/xds_k8s_test_driver/framework/xds_k8s_testcase.py

```diff
@@ -480,13 +480,42 @@ def initKubernetesClientRunner(self) -> KubernetesClientRunner:
 
     def tearDown(self):
         logger.info('----- TestMethod %s teardown -----', self.id())
+        logger.debug('Getting pods restart times')
+        client_restarts: int = 0
+        server_restarts: int = 0
+        try:
+            client_restarts = self.client_runner.get_pod_restarts(
+                self.client_runner.deployment)
+            server_restarts = self.server_runner.get_pod_restarts(
+                self.server_runner.deployment)
+        except (retryers.RetryError, k8s.NotFound) as e:
+            logger.exception(e)
+
         retryer = retryers.constant_retryer(wait_fixed=_timedelta(seconds=10),
                                             attempts=3,
                                             log_level=logging.INFO)
         try:
             retryer(self.cleanup)
         except retryers.RetryError:
             logger.exception('Got error during teardown')
+        finally:
+            # Fail if any of the pods restarted.
+            self.assertEqual(
+                client_restarts,
+                0,
+                msg=(
+                    'Client pods unexpectedly restarted'
+                    f' {client_restarts} times during test. '
+                    'In most cases, this is caused by the test client app crash.'
```

Comment:
Latest internal formatters do that. I know I was doing the opposite before, but I really like this new way. It makes a natural indent - which makes it much clearer this is a continuation of the previous line. 

```suggestion
                    f' {client_restarts} times during test.'
                    ' In most cases, this is caused by the test client app crash.'
```